### PR TITLE
Js restructure weeks 2,3,4 and 5

### PR DESCRIPTION
**Pull request by benna100**
Dont merge before both have taken a look at it!

### CODE REVIEW

File: JavaScript2/readme.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
...
+| 4.   | • JSON <br>• Array Manipulation <br>• Structuring code files | [Preparation](/JavaScript2/Week4/preparation.md)                                                                                  | [Homework](/JavaScript2/Week4/homework.md)                | [Review](/JavaScript2/Week4/review.md)                                               |
+| 5.   | • First Git Session<br>• Events<br>• Callbacks <br>• XHTTP Requests <br>• API calls                                                                                                           | [Preparation](/JavaScript2/Week5/preparation.md)                                                                                  | [Homework](/JavaScript2/Week5/homework.md)   | [Review](/JavaScript2/Week5/review.md)                                                                   |
```
Comment (chklueter) [inline JavaScript2/readme.md]: Are we doing another first and second GIT session though?

File: JavaScript2/Week5/homework.md [hunk 2]
```diff
@@ -6,7 +6,6 @@
 All share a video or a resource (this can be a drawing, article or a pod cast) that was helpful for you the last few weeks with learning JavaScript. Please share this in the channel of your class in Slack. Also write as small note about what the resource i about and why you think it's so helpful (you can share more than one if you like).
```
Comment (chklueter) [inline JavaScript2/Week5/homework.md]: should maybe refer to #teach-yourself channel? Just so people get in the habit of using dedicated channels? Or what do you think?

**Review by chklueter — APPROVED** [review]

**Comment by chklueter** [comment]
Looks fine to me - I of course can't say much about the technical details, but otherwise only made some small comments about details in the text parts :D

File: JavaScript2/readme.md (hunk 1 above)
Comment (benna100) [reply JavaScript2/readme.md]: No, well spotted

File: JavaScript2/Week5/homework.md (hunk 2 above)
Comment (benna100) [reply JavaScript2/Week5/homework.md]: Yup, good idea

File: JavaScript1/Week3/readme.md [hunk 3]
```diff
@@ -1,13 +1,14 @@
-
-# Reading material for the third lecture:
+# Week 3
 
 ## Learning goals
 - [ ] Functions
-- [ ] Code flow (order of execution)
+  - Order of execution
+  - The call stack
+  - Function as a parameter
```
[the remark this answers is not in the document]
Comment (benna100) [reply JavaScript1/Week3/readme.md]: Cool, we will postpone that

File: JavaScript1/Week3/preparation.md [hunk 4]
```diff
@@ -1,97 +1,16 @@
...
-    Notes: for the lectures and homework you only need to know about these events: `click`, `change`, `keyup` and `load`.
-    
-    _You can skip the following sections (but come to these sections when doing the React module):_
-
-    - Touch events
-    - Scroll events
-    - Focus events
-    - Events and the Event Loop
-    - Debouncing
+- Chapter 13: [JavaScript and the Browser](http://eloquentjavascript.net/13_browser.html) (10 min)
```
[the remark this answers is not in the document]
Comment (benna100) [reply JavaScript1/Week3/preparation.md]: Yeah i agree. I have removed some of the reading links

File: JavaScript2/Week4/review.md [hunk 5]
```diff
@@ -1,13 +1,154 @@
...
+const firstArray = ['milk', 'sugar', 'butter', 'flour', 'eggs'];
+
+console.log(firstArray.indexOf('sugar')); // 1
+
+console.log(firstArray.slice(0, 2)); // ['milk', 'sugar']
+
+console.log(firstArray.reverse()); // [, 'eggs', 'flour', 'butter' ,'sugar', 'milk']
 ```
 
 ## Review by Pankaj
```
[the remark this answers is not in the document]
Comment (benna100) [reply JavaScript2/Week4/review.md]: Yeah im not sure why that is in there. I have removed it.

File: JavaScript1/Week3/review.md [hunk 6]
```diff
@@ -1,164 +1,106 @@
...
-console.log(firstArray) // logs: ['milk', 'sugar', 'butter', 'flour', 'eggs']
-
-firstArray.unshift('bagel');
-console.log(firstArray) // logs: ['bagel', 'milk', 'sugar', 'butter', 'flour', 'eggs']
-
-firstArray.shift();
-console.log(firstArray) // logs: ['milk', 'sugar', 'butter', 'flour', 'eggs']
+function cakeLogger(numberOfCakes) {
+    let cakeString = "";
+    for (let i = 0; i < numberOfCakes; i++) {
```
[the remark this answers is not in the document]
Comment (benna100) [reply JavaScript1/Week3/review.md]: Yup, good idea

**Pull request merged**